cmd/agent/app/builder_test.go

```diff
@@ -21,27 +21,55 @@
 package app
 
 import (
-	"io/ioutil"
+	"errors"
 	"strings"
 	"testing"
 
 	"github.com/stretchr/testify/assert"
 	"github.com/stretchr/testify/require"
 	"github.com/uber/jaeger-lib/metrics"
-	"github.com/uber/tchannel-go"
 	"go.uber.org/zap"
 	"gopkg.in/yaml.v2"
 
-	"github.com/uber/jaeger/pkg/discovery"
 	"github.com/uber/jaeger/thrift-gen/jaeger"
 	"github.com/uber/jaeger/thrift-gen/zipkincore"
 )
 
-func TestConfigFile(t *testing.T) {
+var yamlConfig = `
+ignored: abcd
```

Comment:
Why do we ignore keys? 
I feel that we should fail on startup if there are unnecessary keys in the yaml because it can lead to situations where a typo in the yaml can go unnoticed.